framework/src/application/node/node.ts

```diff
@@ -0,0 +1,695 @@
+	}
+
+	public async bootstrap(): Promise<void> {
+		try {
+			// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+			if (!this._options.genesisBlock) {
+				throw Error('Missing genesis block');
+			}
+
+			if (
+				this._options.forging.waitThreshold >= this._options.constants.blockTime
+			) {
+				throw Error(
+					// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+					`forging.waitThreshold=${this._options.forging.waitThreshold} is greater or equal to genesisConfig.blockTime=${this._options.constants.blockTime}. It impacts the forging and propagation of blocks. Please use a smaller value for forging.waitThreshold`,
+				);
+			}
+
+			this._networkIdentifier = getNetworkIdentifier(
+				this._options.genesisBlock.payloadHash,
+				(this._options.genesisBlock as GenesisBlockInstance)
+					.communityIdentifier,
+			);
+
+			this._sequence = new Sequence({
+				onWarning: (current: number): void => {
+					this._components.logger.warn({ queueLength: current }, 'Main queue');
+				},
+			});
+
+			this._initModules();
+
+			// Prepare dependency
+			const processorDependencies = {
+				networkIdentifier: this._networkIdentifier,
+				chainModule: this._chain,
+				bftModule: this._bft,
+				dposModule: this._dpos,
+				logger: this._logger,
+				constants: this._options.constants,
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+				storage: this._storage,
+			};
+
+			this._processor.register(new BlockProcessorV2(processorDependencies));
+
+			// Deserialize genesis block and overwrite the options
+			this._options.genesisBlock = await this._processor.deserialize(
+				(this._options.genesisBlock as unknown) as BlockJSON,
+			);
+
+			this._channel.subscribe('app:state:updated', (event: EventInfoObject) => {
+				Object.assign(this._applicationState, event.data);
+			});
+
+			this._logger.info('Modules ready and launched');
+			// After binding, it should immediately load blockchain
+			await this._processor.init(this._options.genesisBlock);
+			// Check if blocks are left in temp_blocks table
+			await this._synchronizer.init();
+
+			// Update Application State after processor is initialized
+			await this._channel.invoke('app:updateApplicationState', {
+				height: this._chain.lastBlock.height,
+				lastBlockId: this._chain.lastBlock.id,
+				// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+				maxHeightPrevoted: this._chain.lastBlock.maxHeightPrevoted || 0,
+				blockVersion: this._chain.lastBlock.version,
+			});
+
+			// Deactivate broadcast and syncing during snapshotting process
+			if (!Number.isNaN(parseInt(this._options.rebuildUpToRound, 10))) {
+				await this._rebuilder.rebuild(this._options.rebuildUpToRound);
+				this._logger.info(
+					{
+						rebuildUpToRound: this._options.rebuildUpToRound,
+					},
+					'Successfully rebuild the blockchain',
+				);
+				process.exit(0);
+				return;
+			}
```

Comment:
since this logic is only for rebuild, can we move this to top of bootstrap? that way its much clearer while reading as well.

Reply:
I think other bootstrapping needs to happen before, that's why it's here.
Also, rebuilder will be removed

Reply:
if the option `rebuildUpToRound` exists it doesn't need other bootstrapping info? 
yeah since we remove its ok.